photon-client/src/main/scala/com/linkedin/photon/ml/hyperparameter/ShrinkSearchRange.scala

```diff
@@ -0,0 +1,126 @@
+package com.linkedin.photon.ml.hyperparameter
+import breeze.linalg.{DenseMatrix, DenseVector}
+import com.linkedin.photon.ml.hyperparameter.estimators.GaussianProcessEstimator
+import com.linkedin.photon.ml.hyperparameter.estimators.kernels.Matern52
+import org.apache.commons.math3.random.SobolSequenceGenerator
+import scala.math.{floor, max, min}
+
+/**
+  * An object to shrink search range given prior data.
+  */
+object ShrinkSearchRange {
+
+  /**
+    * Compute the lower bound and upper bound of the new prior config
+    *
+    * @param hyperParams Configurations of hyper-parameters
+    * @param priorJsonString JSON string containing prior observations
+    * @param priorDefault Default values for missing hyper-parameters
+    * @param radius The radius of search range after transformation to [0, 1]
+    * @return A tuple of lower bounds and upper bounds for each hyperparameter. Lower bounds and upper bounds are
+    *         discrete for discrete parameters. Lower bounds and upper bounds are dense vectors.
+    */
+  def getBounds(hyperParams: HyperparameterConfig,
+                priorJsonString: String,
+                priorDefault: Map[String, String],
+                radius: Double,
+                seed: Long = System.currentTimeMillis): (DenseVector[Double], DenseVector[Double]) = {
+
+    val hyperparameterList = hyperParams.names
+    val ranges = hyperParams.ranges
+    val discreteParams = hyperParams.discreteParams
+    val numParams = ranges.length
+
+    // Get a [[Seq]] of (vectorized hyper-parameters, evaluationValue) tuples
+    val hyperparameterPairs = HyperparameterSerialization.priorFromJson(priorJsonString, priorDefault, hyperparameterList)
+
+    // Rescale the hyperparameters to [0,1]
+    val hyperparameterRescaled = VectorRescaling.rescalePriors(hyperparameterPairs, hyperParams)
+
+    // Combine hyperparameters as a dense matrix and evaluation value as a dense vector
+    val (overallPoints, overallEvals) = hyperparameterRescaled.map(x => (x._1.asDenseMatrix, DenseVector(x._2))).reduce(
+      (a, b) => (DenseMatrix.vertcat(a._1, b._1), DenseVector.vertcat(a._2, b._2))
+    )
+
+    // Fit Gaussian process regression model
+    val estimator = new GaussianProcessEstimator(kernel = new Matern52)
+    val model = estimator.fit(overallPoints, overallEvals)
+
+    // Sobol generator
+    val paramDistributions = {
+      val sobol = new SobolSequenceGenerator(numParams)
+      sobol.skipTo((seed % (Int.MaxValue.toLong + 1)).toInt)
+      sobol
+    }
+
+    // Draw candidates from a Sobol generator
+    val candidates = (1 until 1000).foldLeft(DenseMatrix(paramDistributions.nextVector)) { case (acc, _) =>
```

Comment:
It might be a good idea to set this 1000 as a parameter with a default being 1000.